docs/src/main/sphinx/develop/client-protocol.rst

```diff
@@ -0,0 +1,257 @@
+======================
+Trino client REST API
+======================
+
+The Trino client allows users to submit Trino queries and view results.  This
+note documents the REST API used by the Trino client.
+
+HTTP methods
+============
+
+* A ``POST`` to ``/v1/statement`` runs the query string in the ``POST`` body,
+  and returns a JSON document containing the query results.  If there are more
+  results, the JSON document will contain a ``nextUri`` URL attribute.
+* A ``GET`` to the ``nextUri`` attribute returns the next batch of query results.
+* A ``DELETE`` to ``nextUri`` terminates a running query.
+
+Overview of query processing
+============================
+
+A Trino client request is initiated by an HTTP ``POST`` to the endpoint
+``/v1/statement``, with a ``POST`` body consisting of the SQL query string.
+The caller may set header ``X-Trino-User`` to the username for the session,
```

Comment:
may set the HTTP header field X-Trino-User to the username of the session. A number of other optional header fields are documented in the following sections.

Reply:
Changed.